### PR TITLE
15596::Added changes to implement divide and added test case

### PR DESCRIPTION
**Pull request by Meesum1799**
I have made the changes, please have a look.

### CODE REVIEW

**Comment by Meesum1799** [comment]
Close #15596

**Comment by xoiga123** [comment]
@Meesum1799 Hi, thank you for contributing to Ivy :hugs: The PR seems to be good, however I've noticed that `tensor.divide` is simply an alias of `tensor.div` as noted in [the torch docs](https://pytorch.org/docs/stable/generated/torch.divide.html). Can you follow [the guideline for alias implementation](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#frontend-apis), maybe check out what other people have done for other aliases here:

https://github.com/unifyai/ivy/blob/2a62323b721f301ddd063a767edc38da5fe18a3e/ivy/functional/frontends/torch/tensor.py#L1027C3-L1030

Thank you!

**Comment by Meesum1799** [comment]
> @Meesum1799 Hi, thank you for contributing to Ivy 🤗 The PR seems to be good, however I've noticed that `tensor.divide` is simply an alias of `tensor.div` as noted in [the torch docs](https://pytorch.org/docs/stable/generated/torch.divide.html). Can you follow [the guideline for alias implementation](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#frontend-apis), maybe check out what other people have done for other aliases here:
> 
> https://github.com/unifyai/ivy/blob/2a62323b721f301ddd063a767edc38da5fe18a3e/ivy/functional/frontends/torch/tensor.py#L1027C3-L1030
> 
> Thank you!

Hey @xoiga123 Sure, Thanks for the review. I will incorporate the feedback and let you know.

**Comment by Meesum1799** [comment]
> > @Meesum1799 Hi, thank you for contributing to Ivy 🤗 The PR seems to be good, however I've noticed that `tensor.divide` is simply an alias of `tensor.div` as noted in [the torch docs](https://pytorch.org/docs/stable/generated/torch.divide.html). Can you follow [the guideline for alias implementation](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#frontend-apis), maybe check out what other people have done for other aliases here:
> > https://github.com/unifyai/ivy/blob/2a62323b721f301ddd063a767edc38da5fe18a3e/ivy/functional/frontends/torch/tensor.py#L1027C3-L1030
> > Thank you!
> 
> Hey @xoiga123 Sure, Thanks for the review. I will incorporate the feedback and let you know.

Hey @xoiga123  I have incorporated the feedback on it. Please have a look on it.

**Comment by xoiga123** [comment]
@Meesum1799 Well done! However, is there any reason that you didn't follow the already existing test of `test_torch_instance_div`?

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -8273,6 +8273,41 @@ def test_torch_instance_multiply(
...
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        init_all_as_kwargs_np={
+            "data": x[0],
+        },
+        method_input_dtypes=input_dtype,
+        method_all_as_kwargs_np={
+            "other": x[1],
+            "rounding_mode": None,
```
Comment (xoiga123) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8304]: Is this always `None`?

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -8308,5 +8343,6 @@ def test_torch_instance_norm(
         frontend_method_data=frontend_method_data,
         init_flags=init_flags,
         method_flags=method_flags,
+
```
Comment (xoiga123) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8346]: Please revert this blank line :cloud:

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -8273,6 +8273,41 @@ def test_torch_instance_multiply(
...
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        init_all_as_kwargs_np={
+            "data": x[0],
+        },
+        method_input_dtypes=input_dtype,
+        method_all_as_kwargs_np={
+            "other": x[1],
```
Comment (xoiga123) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8303]: What happens if `x[1] == 0`?

**Review by xoiga123 — CHANGES_REQUESTED** [review]

**Comment by xoiga123** [comment]
add_frontend_checklist

**Comment by xoiga123** [comment]
add_frontend_checklist

**Comment by Meesum1799** [comment]
@xoiga123 Hey,
I have resolved the issues you have mentioned and now I think all the edge cases got handled so please have a look.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 4]
```diff
@@ -7888,6 +7888,44 @@ def test_torch_instance_div_(
     )
 
 
+# divide
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch_frontend.div",
```
Comment (xoiga123) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:7894]: We don't use `torch_frontend` in `init_tree`.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 5]
```diff
@@ -7888,6 +7888,44 @@ def test_torch_instance_div_(
...
+    method_name="divide",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=2,
+    ),
```
Comment (xoiga123) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:7899]: Please generate `rounding_mode` in the helper.

**Review by xoiga123 — CHANGES_REQUESTED** [review]
Please run the test and make sure it passes :+1:

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 5 above)
Comment (Meesum1799) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:7899]: @xoiga123 Sure, Thanks for the feedback. I will update you as soon as possible.

**Comment by Meesum1799** [comment]
Hey @xoiga123,
I have incorporated the feedback please have a look.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**